components/calendar/src/types.rs

```diff
@@ -536,42 +536,44 @@ impl Month {
 #[derive(Copy, Clone, Debug, PartialEq)]
 #[non_exhaustive]
 pub struct MonthInfo {
-    /// The month number in this given year. For calendars with leap months, all months after
+    /// The ordinal month number in this given year. For calendars with leap months, all months after
     /// the leap month will end up with an incremented number.
     ///
-    /// In general, prefer using the month code in generic code.
+    /// In general, prefer using [`Month`]s in generic code.
     pub ordinal: u8,
 
-    /// The month code, used to distinguish months during leap years.
+    /// The [`Month`], used to distinguish months during leap years.
     ///
     /// Round-trips through `Date` constructors like [`Date::try_new_from_codes`] and [`Date::try_from_fields`].
     ///
     /// This follows [Temporal's specification](https://tc39.es/proposal-intl-era-monthcode/#table-additional-month-codes).
-    /// Months considered the "same" have the same code: This means that the Hebrew months "Adar" and "Adar II" ("Adar, but during a leap year")
-    /// are considered the same month and have the code M05.
+    /// Months considered the "same" are equal: This means that the Hebrew months "Adar" and "Adar II" ("Adar, but during a leap year")
+    /// are considered the same month, `Month::new(6)`.
     ///
     /// [`Date::try_new_from_codes`]: crate::Date::try_new_from_codes
     /// [`Date::try_from_fields`]: crate::Date::try_from_fields
-    pub standard_code: MonthCode,
-
-    /// Same as [`Self::standard_code`] but with invariants validated.
-    pub(crate) standard: Month,
+    pub standard: Month,
 
-    /// A month code, useable for formatting.
+    /// A [`Month`] useable for formatting.
     ///
     /// Does NOT necessarily round-trip through `Date` constructors like [`Date::try_new_from_codes`] and [`Date::try_from_fields`].
     ///
-    /// This may not necessarily be the canonical month code for a month in cases where a month has different
+    /// This may not necessarily be the canonical month for a month in cases where a month has different
     /// formatting in a leap year, for example Adar/Adar II in the Hebrew calendar in a leap year has
-    /// the standard code M06, but for formatting specifically the Hebrew calendar will return M06L since it is formatted
-    /// differently.
+    /// the standard month `Month::new(6)`, but for formatting specifically the Hebrew calendar will return
+    /// `Month::leap(6)`, since it is formatted differently.
     ///
     /// [`Date::try_new_from_codes`]: crate::Date::try_new_from_codes
     /// [`Date::try_from_fields`]: crate::Date::try_from_fields
-    pub formatting_code: MonthCode,
+    pub formatting: Month,
+
+    /// The [`Month::code()`] of [`Self::standard`].
+    #[deprecated(since = "2.2.0", note = "use `standard.code()")]
+    pub standard_code: MonthCode,
```

Comment:
too bad we don't have getters

(but there are good reasons Rust doesn't do that)